integrationTests/vm/arwen/arwenVM_test.go

```diff
@@ -0,0 +1,235 @@
+
+	scCode, err := ioutil.ReadFile("./fib_arwen.wasm")
+	assert.Nil(t, err)
+
+	scCodeString := hex.EncodeToString(scCode)
+
+	tx := vm.CreateTx(
+		t,
+		senderAddressBytes,
+		vm.CreateEmptyAddress().Bytes(),
+		senderNonce,
+		transferOnCalls,
+		gasPrice,
+		gasLimit,
+		scCodeString+"@"+hex.EncodeToString(factory.ArwenVirtualMachine),
+	)
+
+	txProc, accnts, _ := vm.CreatePreparedTxProcessorAndAccountsWithVMs(t, senderNonce, senderAddressBytes, senderBalance)
+
+	err = txProc.ProcessTransaction(tx, round)
+	assert.Nil(t, err)
+
+	_, err = accnts.Commit()
+	assert.Nil(t, err)
+
+	expectedBalance := big.NewInt(99999700)
+	fmt.Printf("%s \n", hex.EncodeToString(expectedBalance.Bytes()))
+
+	vm.TestAccount(
+		t,
+		accnts,
+		senderAddressBytes,
+		senderNonce+1,
+		expectedBalance)
+}
+
+func Benchmark_VmDeployWithFibbonacciAndExecute(b *testing.B) {
+	runWASMVMBenchmark(b, "./fib_arwen.wasm", b.N, 32)
+}
+
+func Benchmark_VmDeployWithCPUCalculateAndExecute(b *testing.B) {
+	runWASMVMBenchmark(b, "./cpucalculate_arwen.wasm", b.N, 8000)
+}
+
+func Benchmark_VmDeployWithStringConcatAndExecute(b *testing.B) {
+	runWASMVMBenchmark(b, "./stringconcat_arwen.wasm", b.N, 10000)
+}
+
+func runWASMVMBenchmark(tb testing.TB, fileSC string, numRun int, testingValue uint64) {
+	ownerAddressBytes := []byte("12345678901234567890123456789012")
+	ownerNonce := uint64(11)
+	ownerBalance := big.NewInt(100000000)
+	round := uint64(444)
+	gasPrice := uint64(1)
+	gasLimit := uint64(100000)
+	transferOnCalls := big.NewInt(1)
+
+	scCode, err := ioutil.ReadFile(fileSC)
+	assert.Nil(tb, err)
+
+	scCodeString := hex.EncodeToString(scCode)
+
+	tx := &transaction.Transaction{
+		Nonce:     ownerNonce,
+		Value:     transferOnCalls,
+		RcvAddr:   vm.CreateEmptyAddress().Bytes(),
+		SndAddr:   ownerAddressBytes,
+		GasPrice:  gasPrice,
+		GasLimit:  gasLimit,
+		Data:      scCodeString + "@" + hex.EncodeToString(factory.ArwenVirtualMachine),
+		Signature: nil,
+		Challenge: nil,
+	}
+
+	txProc, accnts, blockchainHook := vm.CreatePreparedTxProcessorAndAccountsWithVMs(tb, ownerNonce, ownerAddressBytes, ownerBalance)
+	scAddress, _ := blockchainHook.NewAddress(ownerAddressBytes, ownerNonce, factory.ArwenVirtualMachine)
+
+	err = txProc.ProcessTransaction(tx, round)
+	assert.Nil(tb, err)
+
+	_, err = accnts.Commit()
+	assert.Nil(tb, err)
+
+	alice := []byte("12345678901234567890123456789111")
+	aliceNonce := uint64(0)
+	_ = vm.CreateAccount(accnts, alice, aliceNonce, big.NewInt(10000000000))
+
+	for i := 0; i < numRun; i++ {
+		tx = &transaction.Transaction{
+			Nonce:     aliceNonce,
+			Value:     big.NewInt(0).SetUint64(testingValue),
+			RcvAddr:   scAddress,
+			SndAddr:   alice,
+			GasPrice:  0,
+			GasLimit:  5000,
+			Data:      "_main",
+			Signature: nil,
+			Challenge: nil,
+		}
+
+		startTime := time.Now()
+		err = txProc.ProcessTransaction(tx, round)
+		elapsedTime := time.Since(startTime)
+		fmt.Printf("time elapsed full process %s \n", elapsedTime.String())
```

Comment:
I would not do print inside a benchmark. 
There is no need to calculate the time for processTransaction manually, you can benchmark just ProcessTransaction separately if you are not interested in commit as well

Reply:
deleted and changed. I still like the print variant better.